Makefile

```diff
@@ -1,6 +1,29 @@
+UNAME_S ?= $(shell uname -s)
+UNAME_M ?= $(shell uname -m)
+FFMPEG_PREFIX ?= $(shell pwd)/ffmpeg/$(UNAME_S)-$(UNAME_M)
+FFMPEG_OPTS ?= --prefix=$(FFMPEG_PREFIX) \
+               --enable-static \
+               --disable-shared \
+               --disable-programs \
+               --disable-doc \
+               --disable-everything \
+               --enable-decoder=h264 \
+               --enable-decoder=hevc \
+               --enable-swscale
+CGO_LDFLAGS := -L$(FFMPEG_PREFIX)/lib
+ifeq ($(UNAME_S),Linux)
+ifneq ($(shell find / -name libjpeg.a 2> /dev/null),)
+	CGO_LDFLAGS := "$(CGO_LDFLAGS) -l:libjpeg.a"
```

Comment:
Only statically links `libjpeg.a` if found on system.

Reply:
Why only if found? Isn't it a requirement? 
Also, is this the recommended way to find a `.a` file? isn't pkg-config the common way to find C/C++ deps on linux?

Reply:
- Added that check in so that build will work on macos or linux without the static build
    - apt package: https://libjpeg-turbo.org/Downloads/YUM
    - macos uses `jpeg-turbo` homebrew forumula. https://formulae.brew.sh/formula/jpeg-turbo 
        - `.a` actually available on macos
```
ls /opt/homebrew/Cellar/jpeg-turbo/3.0.1/lib 
cmake                    libjpeg.dylib            libturbojpeg.dylib
libjpeg.8.3.2.dylib      libturbojpeg.0.3.0.dylib pkgconfig
libjpeg.8.dylib          libturbojpeg.0.dylib
libjpeg.a                libturbojpeg.a
```
- `pc` file links to the shared object, so have to specify link to the `.a`.
```
cat /usr/lib/x86_64-linux-gnu/pkgconfig/libjpeg.pc
prefix=/usr
exec_prefix=${prefix}
libdir=${prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include

Name: libjpeg
Description: JPEG image codec
Version: 
Libs: -L${libdir} -ljpeg
Cflags: -I${includedir}
```
- Do not like this find check, it is really slow on macos...
    - I think we can make the assumption that .a is available since we get it on macos as well.
    - Will refactor this so libjpeg static link also works on macos and find check is removed.

Reply:
Ok, currently:
- On linux, static link to libjpeg.
- On macos, dynamic link to libjpeg.
- FFmpeg static link works on both.